crates/driver/src/infra/notify/notification.rs

```diff
@@ -18,4 +18,16 @@ pub enum Kind {
     NonBufferableTokensUsed,
     /// Solver don't have enough balance to submit the solution onchain.
     InsufficientBalance,
+    /// Result of winning solver trying to settle the transaction onchain.
+    Settled(SettleKind),
+}
+
+#[derive(Debug)]
+pub enum SettleKind {
+    /// Winning solver settled successfully transaction onchain.
+    Settled(eth::TxId),
+    /// Winning solver mined reverted transaction.
+    Reverted(eth::TxId),
+    /// Winning solver failed to settle the transaction onchain.
+    Failed,
```

Comment:
It seems somewhere we should have an error case specifically for the driver aborting execution because the simulation started to revert.
That sounds like it would be covered by the Failed case but IMO this error is so common that it deserves its own error variant.

Reply:
Makes sense, added.